packages/grid/data-grid/src/tests/toolbar.DataGrid.test.tsx

```diff
@@ -42,65 +50,125 @@ describe('<DataGrid /> - Toolbar', () => {
+      fireEvent.click(getByText('Comfortable'));
 
-    // @ts-expect-error need to migrate helpers to TypeScript
-    expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
-      maxHeight: `${Math.floor(rowHeight * COMFORTABLE_DENSITY_FACTOR)}px`,
+      // @ts-expect-error need to migrate helpers to TypeScript
+      expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
+        maxHeight: `${Math.floor(rowHeight * COMFORTABLE_DENSITY_FACTOR)}px`,
+      });
     });
-  });
 
-  it('should increase grid density even if toolbar is not enabled', () => {
-    const rowHeight = 30;
-    render(
-      <div style={{ width: 300, height: 300 }}>
-        <DataGrid {...baselineProps} rowHeight={rowHeight} density="compact" />
-      </div>,
-    );
-
-    // @ts-expect-error need to migrate helpers to TypeScript
-    expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
-      maxHeight: `${Math.floor(rowHeight * COMPACT_DENSITY_FACTOR)}px`,
+    it('should increase grid density even if toolbar is not enabled', () => {
+      const rowHeight = 30;
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} rowHeight={rowHeight} density="compact" />
+        </div>,
+      );
+
+      // @ts-expect-error need to migrate helpers to TypeScript
+      expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
+        maxHeight: `${Math.floor(rowHeight * COMPACT_DENSITY_FACTOR)}px`,
+      });
+    });
+
+    it('should decrease grid density even if toolbar is not enabled', () => {
+      const rowHeight = 30;
+      render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} rowHeight={rowHeight} density="comfortable" />
+        </div>,
+      );
+
+      // @ts-expect-error need to migrate helpers to TypeScript
+      expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
+        maxHeight: `${Math.floor(rowHeight * COMFORTABLE_DENSITY_FACTOR)}px`,
+      });
     });
   });
 
-  it('should decrease grid density even if toolbar is not enabled', () => {
-    const rowHeight = 30;
-    render(
-      <div style={{ width: 300, height: 300 }}>
-        <DataGrid {...baselineProps} rowHeight={rowHeight} density="comfortable" />
-      </div>,
-    );
-
-    // @ts-expect-error need to migrate helpers to TypeScript
-    expect(screen.getAllByRole('row')[1]).toHaveInlineStyle({
-      maxHeight: `${Math.floor(rowHeight * COMFORTABLE_DENSITY_FACTOR)}px`,
+  describe('<DataGrid /> - Column selector', () => {
+    it('should hide "id" column when hiding it from the column selector', () => {
+      const { getByText } = render(
+        <div style={{ width: 300, height: 300 }}>
+          <DataGrid {...baselineProps} showToolbar />
+        </div>,
+      );
+
+      expect(getColumnHeaders()).to.deep.equal(['id', 'brand']);
+
+      fireEvent.click(getByText('Columns'));
+      fireEvent.click(document.querySelectorAll('[role="tooltip"] [name="id"]')[0]);
```

Comment:
A related not. Is it expected to have a role tooltip on the Popper we display? We don't have any documentation about accesses the accessibility on the component's page https://next.material-ui.com/components/popper/. I wonder if it would make sense to add one cc @eps1lon.

Reply:
@DanailH Why are we using querySelectorAll over querySelector?

Reply:
> @DanailH Why are we using querySelectorAll over querySelector if we only query one element?

That's my bad, I must have been testing different things and forgot about that, I'll fix it.

Regarding the Popper missing a role -> that would have been my prefered method of selecting it

Reply:
One would have to look into whether this makes sense. For example, https://next.material-ui.com/components/popper/#scroll-playground probably is not a tooltip. It would be confusing to have a Popper and Tooltip when both have `role="tooltip"`